repos/system_upgrade/el7toel8/actors/initramdiskgenerator/libraries/initramgen.py

```diff
@@ -26,92 +34,115 @@ def copy_dracut_modules(context, modules):
     Copies our dracut modules into the target userspace.
     """
     try:
-        shutil.rmtree(context.full_path('/dracut'))
+        shutil.rmtree(context.full_path(DRACUT_DIR))
     except EnvironmentError:
         pass
     for module in modules:
         try:
-            context.copytree_to(module.module_path, os.path.join('/dracut', os.path.basename(module.module_path)))
+            context.copytree_to(module.module_path, os.path.join(DRACUT_DIR, os.path.basename(module.module_path)))
         except shutil.Error as e:
             api.current_logger().error('Failed to copy dracut module "{name}" from "{source}" to "{target}"'.format(
-                name=module.name, source=module.module_path, target=context.full_path('/dracut')), exc_info=True)
+                name=module.name, source=module.module_path, target=context.full_path(DRACUT_DIR)), exc_info=True)
             raise StopActorExecutionError(
                 message='Failed to install dracut modules required in the initram. Error: {}'.format(str(e))
             )
 
 
+@suppress_deprecation(UpgradeDracutModule)
 def _get_dracut_modules():
-    return list(api.consume(UpgradeDracutModule))
+    return set(api.consume(UpgradeDracutModule))
 
 
-def install_initram_deps(context):
+def _install_initram_deps(packages):
     used_repos = api.consume(UsedTargetRepositories)
     target_userspace_info = next(api.consume(TargetUserSpaceInfo), None)
 
-    packages = set()
-    for message in api.consume(RequiredUpgradeInitramPackages):
-        packages.update(message.packages)
     dnfplugin.install_initramdisk_requirements(
-        packages=packages, target_userspace_info=target_userspace_info, used_repos=used_repos)
+        packages=packages,
+        target_userspace_info=target_userspace_info,
+        used_repos=used_repos)
 
 
-def _install_files(context):
+# duplicate of _copy_files fro userspacegen.py
```

Comment:
could you please elaborate on why is the duplicate needed?

Reply:
Because it is not function I would like to present publicly for other actors and basically the same has to happen inside two actors. is it answer to the question? If not, I need to remind myself a little bit better when I return to this PR to finish it.